src/changelogs/fetch_changelogs.py

```diff
@@ -4,27 +4,79 @@
 import feedparser
 from bs4 import BeautifulSoup
 from urllib import request
+from utils import json_utils
+from typing import TypedDict
+
+
+class Changelog(TypedDict):
+    date: str
+    link: str
+
+
+class ChangelogLine(TypedDict):
+    Description: str
+    Tags: list[str]
 
 
 class ChangelogFetcher:
+    """
+    Fetches changelogs from the deadlock forums and parses them into a dictionary
+    """
+
+    # Hero lab changelogs need to be added manually to
+    # ./input-data/raw-changelogs following the
+    # naming convention 'herolab_2024_10_29.txt'
+    # see the referenced file for example formatting
+
+    # Then they need to be added to
+    # /output-data/changelogs/changelogs.json following
+    # the same naming convention and formatted the same
+    # as other changelogs, but with "forum_id": null
+
     def __init__(self):
-        self.changelogs_by_date = {}
+        self.changelog_lines: dict[str, ChangelogLine] = {}
+        self.changelogs: dict[str, Changelog] = {}
 
     def get_rss(self, rss_url, update_existing=False):
         self.RSS_URL = rss_url
         self._fetch_forum_changelogs(update_existing)
-        return self.changelogs_by_date
+
+        return self.changelog_lines
 
     def get_txt(self, changelog_path):
         self._process_local_changelogs(changelog_path)
-        return self.changelogs_by_date
+        return self.changelog_lines
 
     def changelogs_to_file(self, output_dir):
-        for date, changelog in self.changelogs_by_date.items():
-            os.makedirs(output_dir, exist_ok=True)
-            with open(output_dir + f'/{date}.txt', 'w', encoding='utf8') as f_out:
+        # Write raw changelog lines to files
+        for version, changelog in self.changelog_lines.items():
+            raw_output_dir = os.path.join(output_dir, 'raw')
+            os.makedirs(raw_output_dir, exist_ok=True)
+            with open(raw_output_dir + f'/{version}.txt', 'w', encoding='utf8') as f_out:
                 f_out.write(changelog)
 
+        # Write non-line data for the changelogs to 1 file
+
+        # this file is not overwritten even when update_existing is True
+        # many entries were manually added due to only the first page on the site has rss feed
+        # herolab entries also need to be added manually as they are not on the forum at all
+
```

Comment:
Struggling to see how this comment block pertains to this function

Reply:
removed the block and distributed parts of the comments to their relevant sections